### DISCUSSION TITLE
Best way to delay validation of empty values for a required field

### DISCUSSION THREAD

**Discussion by zayenz** [airjp73/rvf#79] [Q&A] [answered]
I feel that there is a qualitative difference between validating input that a user has written vs indicating that a field is required. When validating something that the user has written, I want to do it `onBlur` or `onChange`, while for validating that a field is required I typically want to do it `onSubmit`.

Currently, I am checking in my input component if the form has been submitted and what the current input value is to avoid showing validation errors for empty fields too early. Is there a better or more idiomatic way to do this?

**Comment by airjp73** [comment, marked as answer]
If that's the behavior you want, then you'll probably need to handle your validation manually and skip `getInputProps`. All the state we track for validation is available for you to check / update directly.

```ts
const {
  validate,
  clearError,
  error,
  touched,
  setTouched,
  defaultValue,
} = useField(name)

return (
  <input
    name={name}
    defaultValue={defaultValue}
    onChange={e => {
      if (e.target.value) validate()
    }}
    onBlur={() => validate()}
  />
)
```

**Reply by zayenz** [reply]
Nice, thanks. I hadn't considered just replacing the onChange and onBlur. I'll experiment with various options.
